drivers: sensor: icm42670: supports icm42670-P and icm42670-S

afontaine-invn opened the pull request:
dts: supports icm42670-P/-S
samples: sensor: tdk_apex: TDK APEX generic sample
samples: sensor: 6DOF motion DRDY generic sample
module: HAL TDK module

Adds official TDK Invensense Inc. driver in TDK HAL  module for icm42670-P/-S sensor.
Adds SPI and I2C interface support.
Adds APEX features, such as Pedometer, Tilt detection, Wake on Motion
and Significant Motion Detector.
Add generic samples supporting this driver.
Integration module test successful with https://github.com/tdk-invn-oss/zephyr.hal_tdk  
Validated with custom setup: nrf52dk_nrf52832 + icm42670-P/-S EVB board
Build ok by running:
west twister -T samples\sensor\tdk_apex-T samples\sensor\6dof_motion_drdy -p nrf52dk/nrf52832

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

gmarull reviewed (changes requested):
please, check Zephyr contribution guidelines. There seem to be PDFs and binary files in this PR.

teburd reviewed (commented):
An initial pass here (and this is large so I'll need time to look at it again) with some high level notable comments but...

I love that this is being submitted as a PR from you, with a TDK email address. I hope this is just the first of many to be frank.

That said there's some issues that will need to be addressed. Blobs are problematic, pdf's are problematic, rewriting the register map but not using the zephyr macros for describing masks and manipulating fields seems like a step backwards. Using the existing trigger API requiring a thread or work queue also feels like a step backwards, but that's not on you.

Is the code under the imu directory here something from a TDK software devkit? If so it might be worth considering looking at how ST packages and uses stmemsc for ST sensor drivers. It does not need to live or be managed then by Zephyr but can be managed entirely but TDK in something like a hal_tdk git repository.

carlescufi commented:
Thanks for the PR! Please take a look at [our commit message guidelines](https://docs.zephyrproject.org/latest/contribute/guidelines.html#commit-message-guidelines) to find out how to format your commit messages, and at [our contribution workflow](https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-workflow) to understand how to update your Pull Request.

afontaine-invn commented:
Hi @teburd, @carlescufi, thank you very much for your review! You point many subjects to address, Find my answers below: 

- Regarding PDF and blobs I'll remove it for now. We though exposing the AML library (featuring pointing, gesture recognition, ...) but it seems not feasible and compatible with Zephyr rules.
- Regarding the trigger API requiring a thread or work queue I kept it configurable by KConfig and I kept the option to not configure any trigger. I can put it as default.

- Regarding the possibility to use a hal_tdk git repository to put the imu driver directory, we look at how st handles it with @gjabouley-invn and we'll like the idea. Could you give us more details on how to put it in place? And if there is any rules or guidelines to manage it over the time?

- And just as you know I'm not the only one to push with a TDK address lately, @
rbuisson-invn also updated ICP drivers with official ones. Find the PR https://github.com/zephyrproject-rtos/zephyr/pull/69282

decsny commented on dts/bindings/sensor/invensense,icm42670-i2c.yaml:
any DT binding property with a default value needs justification of the default value in the description https://docs.zephyrproject.org/latest/build/dts/bindings-upstream.html

afontaine-invn replied:
I updated the DT bindings with default description as required..

afontaine-invn commented on drivers/sensor/tdk/icm42670/invn.algo.sw-integration.aml-42670s-gcc-arm-none-eabi-cm4-fpu-1.6.0/include/invn_algo_aml.h:
```diff
@@ -0,0 +1,177 @@
+/*
+ * $License:
```
This file has been removed

afontaine-invn commented on samples/sensor/icm42670/data_stream/src/main.c:
```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2024 TDK Invensense
```
Sample moved to 6dof_motion_drdy as you suggested.

afontaine-invn commented on samples/sensor/icm42670/apex/src/main.c:
```diff
@@ -0,0 +1,153 @@
+ * Get a device structure from a devicetree node with compatible
+ * "invensense,icm42670". (If there are multiple, just pick one.)
+ */
+static const struct device *get_icm42670_device(void)
```
Sample moved to tdk_apex as you suggested.

afontaine-invn commented on drivers/sensor/tdk/icm42670/invn.algo.sw-integration.aml-42670s-gcc-arm-none-eabi-cm4-fpu-1.6.0/include/invn_algo_aml.h:
```diff
@@ -0,0 +1,177 @@
+ * AML output structure (calibrated sensor offsets, pointing output, swipe gestures detected and
+ * quaternion)  \ingroup AML
+ */
+typedef struct {
```
This file has been removed

decsny commented on dts/bindings/sensor/invensense,icm42670-common.yaml:
```diff
@@ -0,0 +1,138 @@
+
+  power-mode:
+    type: string
+    default: "low noise"
```
missing justification

afontaine-invn replied:
Thanks for the comment. I added the default value justification for this property also.

teburd commented:
Sensor WG 04/22/2024

Asked the question around pros/cons of using sensor vendor supplied HALs:

Pros:
* Potentially more vendor involvement, with more drivers being pushed to the tree

Potential Cons:
* Unclear how much benefit the HAL provides, e.g. abstracting bus reads/writes is already doable within Zephyr today.
* Unclear how the maintainer ship burden increases when using HALs, who maintains this code long term and moves it forward as sensor APIs evolve and change.
* Unclear how we move this code to the asynchronous sensor_read API which is the future of sensors to better support more use cases

afontaine-invn commented:
Hi @teburd ,
Thanks for the updates on the tdk_hal repository issue.
Are there any blocking points or any changes expected to move forward with the PR?

teburd commented:
> Hi @teburd , Thanks for the updates on the tdk_hal repository issue. Are there any blocking points or any changes expected to move forward with the PR?

No blocking issues from me no, I expect the TSC will approve hal_tdk barring any major concerns, and then this can go in (as it depends on hal_tdk). I'd expect some more progress in a week or so as the TSC meets weekly, and our sensor maintainer (Maureen Helm at Analog) is traveling this week

gmarull commented on drivers/sensor/tdk/icm42670/Invn/InvError.h:
```diff
@@ -0,0 +1,40 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @defgroup InvError Error code
+ *	@brief    Common error code
+ *
+ *	@ingroup EmbUtils
+ *	@{
+ */
```
this file still doesn't follow Zephyr conventions (doxygen not used in internal headers, UpperCase naming, etc.)

afontaine-invn replied:
Here I removed the @ingroup comments not relevant in this integration. Are there other changes you requested?

gmarull commented on drivers/sensor/tdk/icm42670/Invn/InvError.h:
```diff
@@ -0,0 +1,40 @@
+	INV_ERROR_NIMPL = -2,     /**< function not implemented for given arguments */
+	INV_ERROR_TRANSPORT = -3, /**< error occurred at transport level */
+	INV_ERROR_TIMEOUT = -4,   /**< action did not complete in the expected time window */
+	INV_ERROR_SIZE = -5, /**< argument's size is not suitable to complete requested action */
+	INV_ERROR_OS = -6,   /**< error related to OS */
+	INV_ERROR_IO = -7,   /**< error related to IO operation */
+	INV_ERROR_MEM = -9,  /**< not enough memory to complete requested action */
+	INV_ERROR_HW = -10,  /**< error at HW level */
```
not sure what these are? why not use `errno` codes?

afontaine-invn replied:
Thank you for the review. 
This InvError.h file is part of our driver package (that should move to be in hal_tdk module). I removed any use of them in icm42670.c file.

ubieda commented:
@afontaine-invn would you like to join the Sensors WG meeting on monday to touch base on this PR?
It's next monday 7 to 8 AM PT. Here are the details: https://github.com/zephyrproject-rtos/zephyr/wiki/Sensors-Working-Group

ubieda commented:
> Hi @ubieda , Hi @teburd , Following last TSC meeting, I updated the PR to move all TDK drivers in TDK Hal module. I tested the module integration successfully with [tdk-invn-oss/zephyr.hal_tdk](https://github.com/tdk-invn-oss/zephyr.hal_tdk) copied in module/hal/tdk I updated the west.yml as if the TDK HAL module is already a zephyr repository. Let me know if it's ok as it breaks the build. Or if there is another process?

Thanks for following up!

The process for submitting external code (hal_tdk) as a Module is described here: https://docs.zephyrproject.org/latest/contribute/external.html#submission-and-review-process.

From what I see, your next steps would include:
- Create the GH Issue following the template. Within the information filled, you would link this PR as part of the submission (see this reference: #72391).
- Present this module in the next TSC meeting, where voting takes place to decide on its inclusion after review. These meetings take place on Wednesdays from 8 to 9 AM PT (See: https://github.com/zephyrproject-rtos/zephyr/wiki/Technical-Steering-Committee-%28TSC%29).
- Once the module is approved, the Zephyr Project creates the `zephyrproject-rtos/hal_tdk` repository, managed by the Zephyr Project, maintained and kept up to date by the TDK team.
- That unlocks this PR review to finalize, assuming the contents pertintent to Zephyr are here and the external code is offloaded to the HAL (at a glance it already appears to be the case).

Feel free to reach out if you have any further questions.

@teburd @nashif Please, LMK if I missed anything.

afontaine-invn commented:
Thanks for the process reminder!
The issue is opened: https://github.com/zephyrproject-rtos/zephyr/issues/71561
I'll join the meeting as next step!

LucasTambor commented:
@afontaine-invn I just pushed #74418 before I found out about this PR. I tried to be as minimally invasive as possible to enable the I2C bus for the ICM42670. 

This PR addresses all my needs in addition to adding other features. Do you think is better to close my PR, or we can proceed on approving it before yours, since it is smaller and is likely to get in before this one?

ubieda commented:
> Do you think is better to close my PR, or we can proceed on approving it before yours, since it is smaller and is likely to get in before this one?

I'd vote for pursuing this one instead.

ubieda commented:
@LucasTambor Added you to the reviewer's list per https://github.com/zephyrproject-rtos/zephyr/pull/74418#issuecomment-2176687238

afontaine-invn commented:
> Thanks for the process reminder! The issue is opened: #71561 I'll join the meeting as next step!

HI @ubieda , Do you know what I need to do now that the motion is passed on the last 10th July TSC meeting?

afontaine-invn commented:
Hi @nashif , as I understand the tests doesn't pass because of a missing test in CMakelist in hal_tdk, meaning part of the files are included.  https://github.com/zephyrproject-rtos/hal_tdk/pull/1. How can I push this change?

afontaine-invn commented:
Hi @ubieda , as I understand the tdk_hal module I added breaks the test. I would like to fix it https://github.com/zephyrproject-rtos/hal_tdk/pull/1. Could you help reviewing it to unlock the build in this PR?

afontaine-invn commented:
> please, check Zephyr contribution guidelines. There seem to be PDFs and binary files in this PR.

Hi @gmarull , the PDF and binary files has been removed, could you have a look at this PR? Thanks

gmarull commented on dts/bindings/sensor/invensense,icm42670.yaml:
```diff
@@ -42,14 +44,29 @@ properties:
+      1.5625Hz to 400Hz.
+      The default is chosen to support both accelerometer and
+      gyroscope sensors.
+    enum:
+      - "low noise"
+      - "low power"
```
no spaces in enums please

afontaine-invn replied:
I replaced space by -

gmarull commented on dts/bindings/sensor/invensense,icm42670.yaml:
```diff
@@ -64,6 +81,40 @@ properties:
+      - 32
+      - 64
+
+  accel-filt-bw:
+    type: int
+    default: 180
```
units?

afontaine-invn replied:
Unit is Hz, I added it in the comment.

gmarull replied:
property still misses proper suffix

afontaine-invn replied:
I don't get your point. Which suffix is missing for accel-filt-bw?

gmarull commented on dts/bindings/sensor/invensense,icm42670.yaml:
```diff
@@ -76,3 +127,20 @@ properties:
       - 500
       - 250
+
+  gyro-filt-bw:
+    type: int
+    default: 180
+    description: |
```
units?

afontaine-invn replied:
Unit is Hz, I added it in the comment.

gmarull commented on samples/sensor/6dof_motion_drdy/src/main.c:
```diff
@@ -0,0 +1,119 @@
+{
+	const struct device *const dev = DEVICE_DT_GET(DT_ALIAS(6dof_motion_drdy0));
+
+	if (dev == NULL) {
+		/* No such node, or the node does not have status "okay". */
+		printk("\nError: no device found.\n");
+		return NULL;
+	}
```
this is impossible, build will fail if macro doesn't resolve

afontaine-invn replied:
Ok I removed it. I copied this from the bosh,bme280 example, I though it was needed.

gmarull commented on samples/sensor/tdk_apex/VERSION:
```diff
@@ -0,0 +1,5 @@
+VERSION_MAJOR = 1
+VERSION_MINOR = 0
+PATCHLEVEL = 0
+VERSION_TWEAK = 0
+EXTRAVERSION =
```
??

afontaine-invn replied:
I removed the VERSION files. I though these VERSION files were recommended, but I misunderstood the guidelines.

gmarull commented on samples/sensor/tdk_apex/src/main.c:
```diff
@@ -0,0 +1,159 @@
+{
+	const struct device *const dev = DEVICE_DT_GET(DT_ALIAS(tdk_apex_sensor0));
+
+	if (dev == NULL) {
+		/* No such node, or the node does not have status "okay". */
+		printk("\nError: no device found.\n");
+		return NULL;
+	}
```
same

afontaine-invn replied:
Ok I removed it.

afontaine-invn commented:
Hi @gmarull , I did the changes you requested. Could you update the changes requested status if it's ok for you?

afontaine-invn commented:
Hello @gmarull 
I addressed the change as requested; could you update your review status? Is there something else do you expect?

gmarull commented on samples/sensor/6dof_motion_drdy/README.rst:
```diff
@@ -0,0 +1,49 @@
+.. _6dof_motion_drdy:
+
+6DOF Motion Dataready:
+############################################
```
wrong formatting

afontaine-invn replied:
I updated both README for 6dof_motion_drdy and tdk_apex samples.

gmarull commented on samples/sensor/6dof_motion_drdy/boards/nrf52dk_nrf52832_i2c.overlay:
```diff
@@ -0,0 +1,44 @@
+ * Increase native UART speed to report all IMU data at 100Hz.
+ */
+&uart0 {
+	compatible = "nordic,nrf-uarte";
+	status = "okay";
```
why is this needed?

afontaine-invn commented on samples/sensor/tdk_apex/Kconfig:
```diff
@@ -0,0 +1,26 @@
+
+# Enable APEX features support for the IMU device selected
+config TDK_APEX
+	default y
```
Ok, I moved this to the prj.conf file

pdgendt commented on drivers/sensor/tdk/icm42670/Kconfig:
```diff
@@ -9,13 +10,44 @@ menuconfig ICM42670
+	bool "ICM42670-S IMU 6-Axis Motion Tracking with Air Motion Library (AML)"
+endchoice
+
+config TDK_APEX
+	bool "TDK APEX features"
+	default n
```
Omit `default n`, but it feels like this should be a dts/runtime setting. What if you have multiple sensors, and want each sensor to have a different feature?

afontaine-invn replied:
You are right, it's better to put it in device tree. I did the change and the result is cleaner, thanks!

pdgendt commented on drivers/sensor/tdk/icm42670/Kconfig:
```diff
@@ -9,13 +10,44 @@ menuconfig ICM42670
 
 if ICM42670
 
+choice ICM_VARIANT
+	prompt "ICM variant to identify the sensor"
+	help
+	  Select the ICM variant
+config ICM42670P
+	bool "ICM42670-P IMU 6-Axis Motion Tracking"
+config ICM42670S
+	bool "ICM42670-S IMU 6-Axis Motion Tracking with Air Motion Library (AML)"
+endchoice
```
This should go to device tree instead of Kconfig, imagine having multiple variants.

afontaine-invn replied:
Thanks for this comment as well. I did the change and put this option in device tree. It has more sense like this.

afontaine-invn commented:
Thanks for your comment @pdgendt , it was pertinent and I'm happy with the new version with more options in device tree.

afontaine-invn commented:
Hi @gmarull , I think the changes I made match what you expected. Could you review and remove the changes requested that may block the PR ?

afontaine-invn commented:
Hi @MaureenHelm , could we remove the DNM label on this PR ? I did a lot of changes to match what was expected from the Zephyr members comments and I would like to move forward ?

MaureenHelm commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -1,399 +1,752 @@
+			if (event.sensor_mask & (1 << INV_SENSOR_TEMPERATURE)) {
+				data->temp = event.temperature;
+			}
+			/* TODO use a ringbuffer to handle multiple samples in FIFO */
```
Look at implementing the sensor streaming interface with RTIO

MaureenHelm commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -442,133 +793,136 @@ static int icm42670_sample_fetch_temp(const struct device *dev)
+	if (int_status & INT_STATUS_DRDY_DATA_RDY_INT_MASK) {
+		switch (chan) {
+		case SENSOR_CHAN_ALL:
+			res |= icm42670_sample_fetch_accel(dev);
+			res |= icm42670_sample_fetch_gyro(dev);
+			res |= icm42670_sample_fetch_temp(dev);
```
This is another example where you will end up returning something that isn't an errno anymore because you've logically OR-ed together multiple values

afontaine-invn replied:
Change done

MaureenHelm commented on drivers/sensor/tdk/icm42670/icm42670_trigger.h:
```diff
@@ -14,30 +14,30 @@ int icm42670_trigger_set(const struct device *dev, const struct sensor_trigger *
 
 /**
- * @brief initialize the icm42670 trigger system
+@brief initialize the icm42670 trigger system
```
???

afontaine-invn replied:
Sorry, formatting error on my side. Change reverted.

MaureenHelm commented on dts/bindings/sensor/invensense,icm42670.yaml:
```diff
@@ -14,6 +15,12 @@ properties:
       property value should ensure the flags properly describe the
       signal that is presented to the driver.
 
+  is-variant-s:
+    type: boolean
+    description: |
+      Specifies that the driver should adapt to the ICM-42670-S.
+      This affects the chip ID.
```
Use different compatibles instead

afontaine-invn replied:
Thanks for this suggestion, I created different compatible.

MaureenHelm commented on dts/bindings/sensor/invensense,icm42670.yaml:
```
dts: variant, apex, power, and filtering options

Add the power mode, accel and gyro filtering options, variant S
support, and apex features.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>
```

The first line of the commit message is too vague, can you update it to something like:
```dts: bindings: sensor: Add invensense icm42670 properties```

This makes it easier to summarize sensor-related changes for release notes

afontaine-invn replied:
Commit message updated

MaureenHelm commented on samples/sensor/6dof_motion_drdy/Kconfig:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Trigger mode: sample is using interrupt triggering
+choice ICM42670_TRIGGER_MODE
+	default ICM42670_TRIGGER_OWN_THREAD
+endchoice
```
This file is for defining new Kconfig symbols, not overriding existing symbols that were defined elsewhere. You should set this in prj.conf instead.

afontaine-invn replied:
Thanks for this clarification. I removed the Kconfig and put this in prj.conf

MaureenHelm commented on samples/sensor/6dof_motion_drdy/README.rst:
@avisconti can you take a look at this sample?

MaureenHelm commented on samples/sensor/6dof_motion_drdy/boards/nrf52dk_nrf52832_i2c.overlay:
```diff
@@ -0,0 +1,43 @@
+};
+
+/*
+ * Example configuration of a ICM42670 device on i2c0 compatible with an Arduino I2C bus.
```
Is there an arduino shield available with this sensor?

afontaine-invn replied:
No, there is no Arduino shield available for now.

MaureenHelm commented on samples/sensor/tdk_apex/Kconfig:
```diff
@@ -0,0 +1,9 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Enable trigger to get APEX sensor data
+choice ICM42670_TRIGGER_MODE
+	default ICM42670_TRIGGER_OWN_THREAD
+endchoice
```
Ditto

afontaine-invn replied:
Move to prj.conf as requested

MaureenHelm commented on samples/sensor/tdk_apex/README.rst:
```diff
@@ -0,0 +1,134 @@
+
+Make sure the apex feature used is in devicetree, then build and run with:
+
+.. zephyr-app-commands:
```
```suggestion
.. zephyr-app-commands::
```

afontaine-invn replied:
Change done

MaureenHelm commented on west.yml:
```diff
@@ -237,6 +237,11 @@ manifest:
       path: modules/hal/stm32
       groups:
         - hal
+    - name: hal_tdk
+      revision: 16eaeeb54138348fb3227323d7d7c4b94e96ac3d
+      path: modules/hal/tdk
+      groups:
+        - hal
```
The HAL commit should be first in the series, otherwise this PR isn't bisectable

afontaine-invn replied:
Thanks for this comment. I reorder my commits to have the HAL commit first.

afontaine-invn commented on drivers/sensor/tdk/icm42670/Kconfig:
```diff
@@ -1,18 +1,26 @@
 	bool "ICM42670 Six-Axis Motion Tracking Device"
 	default y
 	depends on DT_HAS_INVENSENSE_ICM42670_ENABLED
+	select ZEPHYR_HAL_TDK_MODULE
```
Change done

afontaine-invn commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -1,399 +1,752 @@
+
+	if (val->val1 <= 1600 && val->val1 >= 1) {
+		if (drv_data->accel_hz == 0) {
+			err |= inv_imu_set_accel_frequency(
```
Thanks for the catch. I updated the error management to report only errno's type to the sensor driver API.

afontaine-invn commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -1,399 +1,752 @@
+	data->serif.read_reg = inv_io_hal_read_reg;
+	data->serif.write_reg = inv_io_hal_write_reg;
+	data->serif.max_read = 1024 * 32;
+	data->serif.max_write = 1024 * 32;
```
Change done

afontaine-invn commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -1,399 +1,752 @@
+#endif
+#if ICM42670_BUS_I2C
+	data->serif.serif_type = UI_I2C;
+#endif
```
Thanks for this catch. It's now handled according to the instance in Kconfig.

afontaine-invn commented on drivers/sensor/tdk/icm42670/icm42670.c:
```diff
@@ -1,399 +1,752 @@
+#endif
+	err = inv_imu_init(&data->driver, &data->serif, NULL);
+	if (err < 0) {
+		LOG_DBG("Init failed: %d", err);
```
Change done

sriccardi-invn commented:
HI @MaureenHelm,
Any update on the MR ?
The branch is opened since a lot of time with different reviewers. My team consume is time to rebase everytime.
Do you think we could converge?
We have already a lot of sensor drivers ready to be integrated in tdk_hal.
Thanks.
Regards,
Sebastien Riccardi

afontaine-invn committed:
module: hal_tdk module

 https://github.com/tdk-invn-oss/zephyr.hal_tdk

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn committed:
drivers: sensor: icm42670: supports icm42670-P/-S

Prepare to use official TDK Invensense Inc. driver for icm42670-P/-S
sensor in tdk_hal module. Simplify I2C and SPI transport files.
Driver code moves in hal_tdk module.
Adds APEX features, such as Pedometer, Tilt detection, Wake on Motion
and Significant Motion Detector.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn committed:
dts: bindings: sensor: Add invensense icm42670 properties

Add the power mode, accel and gyro filtering options,
 and apex features. Add -p and -s compatible.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn committed:
samples: sensor: 6DOF motion DRDY generic sample

It reports IMU 6-axis accelerometer and gyroscope data
using DRDY interrupt.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn committed:
samples: sensor: TDK APEX generic sample

It reports Advanced Pedometer and Event Detection features,
such as Pedometer, Tilt detection, Wake on Motion and
Significant Motion Detector. Device tree options.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn committed:
tests: sensor: update sensor compatible icm42670p

Update i2c and spi dtsi test to comply with new compatible icm42670p.

Signed-off-by: Aurelie Fontaine <aurelie.fontaine@tdk.com>

afontaine-invn commented:
Hello @MaureenHelm , Thanks for the approval, we're happy to achieve this step!
I rebased this PR and resolved the conflict in icm42670.c file. Could you approve again and remove the DNM and dev-review label in order to be able to merge?

sriccardi-invn commented:
HI @MaureenHelm ,
I am affraid by the delay to merge a branch.
Everytime we wait, some change have beed done an another side, we have ro rebase, wait a new validation.
This is infinite.
We have a lot of sensors driver ready to be integrated waiting this first merge.
Do you have or see any remaining issue on your side?
Thanks.
Regards,
Sebastien Riccardi

kartben commented:
Thank you for your patience @afontaine-invn @sriccardi-invn and team, and sorry this took so long. This will go in with the next merge batch in just a couple hours.

pdgendt commented:
Every build now prints:

```
CMake Warning at /home/pdgendt/dev/zephyr-main/zephyr/CMakeLists.txt:996 (message):
  No SOURCES given to Zephyr library: ..__modules__hal__tdk

  Excluding target from build.
```

pdgendt commented:
Created https://github.com/zephyrproject-rtos/zephyr/pull/83151 to get rid of the warning.
